russstation/code/modules/client/preferences/species_features/diona.dm

```diff
@@ -0,0 +1,35 @@
+// based heavily off the pod hair setup in TG code | /datum/preference/choiced/pod_hair in pod.dm
+/datum/preference/choiced/diona_hair
+	savefile_key = "feature_diona_hair"
+	savefile_identifier = PREFERENCE_CHARACTER
+	category = PREFERENCE_CATEGORY_FEATURES
+	main_feature_name = "Hairstyle"
+	should_generate_icons = TRUE
+
+/datum/preference/choiced/diona_hair/init_possible_values()
+	var/list/values = list()
+
+	var/icon/diona_head = icon('russstation/icons/mob/mutant_bodyparts.dmi', "diona_head")
+
+	for (var/diona_name in GLOB.diona_hair_list)
+		var/datum/sprite_accessory/diona_hair = GLOB.diona_hair_list[diona_name]
+
+		var/icon/icon_with_hair = new(diona_head)
+		var/icon/icon_adj = icon(diona_hair.icon, "m_diona_hair_[diona_hair.icon_state]_ADJ")
+		// TODO: Needs front icons?
+		// var/icon/icon_front = icon(diona_hair.icon, "m_diona_hair_[diona_hair.icon_state]_FRONT")
+		// icon_adj.Blend(icon_front, ICON_OVERLAY)
```

Comment:
I'm not familiar with "front icons". What's the issue here?

Reply:
I don't think it is an issue; pretty sure it's only needed if there is something sticking out the front of the head. 

Comment is left there (mistakenly) as I could not get the hairs to render on character customization screen. But that might be due to the underlying JavaScript/Typescript UI code (and likely not an issue as they won't be playable like that from what I'm told).

Reply:
I'd like to see if we can get hair working on the customization screen before this gets merged, considering that would be a decent issue for people trying to stylize their character

Reply:
Got it working; will commit soon just double checking to ensure it's all good
![image](https://user-images.githubusercontent.com/665570/164567923-cd9c35c7-f1c9-4c00-8c34-c56a7d2bb2a8.png)